unity/EditorPlugin/PluginEntryPoint.cs

```diff
@@ -226,9 +220,6 @@ private static void InitializeProtocol(Lifetime lifetime, string protocolInstanc
 
     internal static void InitForPluginLoadedFromAssets()
     {
-      if (ourInitialized)
```

Comment:
I think that flag prevent from double initialization during same AppDomain session: if you have package installed and you still have old Editor Plugin in Assets folder

Reply:
If the plugin is loaded from `Assets`, Unity will invoke the class constructor, because it's `[InitializeOnLoad]`. This sets the flag. 

When the package is initialised, it looks to see if the plugin is already in memory, and if so, does nothing, so there won't be double initialisation.

I guess there could be a race condition here - if the package is loaded before the package, then the package would continue initialisation. But that would load the plugin from the IDE as a byte stream rather than a named assembly or file, which means the two `PluginEntryPoint` classes would be different `Type` instances, so the flag wouldn't help, right?

Reply:
I was reading commit one by one and see that you have refactored already it a bit. Yeah, now everything is correct (after reading the last commit). I suppose that previously in one of Rider Packages we were not checking that library was already loaded

Reply:
Cool, thanks. Marking as resolved